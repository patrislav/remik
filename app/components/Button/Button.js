import React, {PropTypes} from 'react'

const propTypes = {
  onClick: PropTypes.func,
  children: PropTypes.node
}

const defaultProps = {
  onClick: () => {},
  children: ''
}

const Button = ({ onClick, children, ...props }) => (
  <button onClick={onClick} {...props}>{children}</button>
)

Button.propTypes = propTypes
Button.defaultProps = defaultProps

export default Button
