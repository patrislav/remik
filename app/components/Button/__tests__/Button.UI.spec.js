import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

import Button from '../Button'

describe('<Button />', () => {
  it('renders correct UI with default props', () => {
    const wrapper = shallow(<Button />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('renders correct UI with provided label', () => {
    const wrapper = shallow(<Button>LABEL</Button>)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
