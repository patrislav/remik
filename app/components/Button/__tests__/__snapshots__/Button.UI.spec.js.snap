exports[`<Button /> renders correct UI with default props 1`] = `
<button
  onClick={[Function]} />
`;

exports[`<Button /> renders correct UI with provided label 1`] = `
<button
  onClick={[Function]}>
  LABEL
</button>
`;
